.github/workflows/build-pr.yml

```diff
@@ -4,19 +4,67 @@ on:
   push:
     branches: [ "master" ]
   pull_request:
-    branches: [ "master" ]
+    branches: 
+      - master
+      - 'feature/**'
 
 jobs:
-  build:
-
+  test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read      # required by actions/checkout
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install poetry
+        poetry install --with test
+    - name: Check tests 
+      run: |
+        poetry run pytest tests/
+  formatting:
```

Comment:
Would it make sense to consider merging formatting and analysis in a single job? So that we don't create environments so many times?
At the same time I guess if all those jobs can take place in parallel, maybe this makes the CI more convenient overall?
No strong opinion either way, just a question.

Reply:
Sure, no strong opinion either. It makes it a bit less redundant.